include/tscore/Extendible.h

```diff
@@ -575,6 +584,17 @@ sizeOf(size_t size = sizeof(Derived_t))
 template <class Derived_t> Extendible<Derived_t>::Extendible()
 {
   ink_assert(ext::details::areFieldsFinalized());
+  // don't call callConstructor until the derived class is fully constructed.
+  ++schema.cnt_constructed;
+}
+
+template <class Derived_t> Extendible<Derived_t>::~Extendible()
+{
+  // assert callConstructors was called.
+  ink_assert(ext_loc);
+  schema.callDestructor(uintptr_t(this) + ext_loc);
```

Comment:
Could you use `std::destroy_at` here?

Reply:
I think not. schema like the type eraser field factory. callDestructor executes functors/lambdas for each field on that extendible block of memory.